transportservice/Services/Transport/EventHandlers.cs

```diff
@@ -100,6 +100,47 @@ public async void Handle(SeatsChangedEvent @event)
         if (queryTransportOption != null)
         {
             queryTransportOption.Seats += @event.ChangeBy;
+
+            if (@event.ChangeBy < 0)
+            {
+                var destination = dbContext.PopularDestinations
+                    .FirstOrDefault(pd => pd.Country == queryTransportOption.ToCountry && pd.City == queryTransportOption.ToCity);
+
+                var type = dbContext.PopularTransportTypes.FirstOrDefault(pdt => pdt.Type == queryTransportOption.Type);
+
+                if (destination == null)
+                {
+                    destination = new PopularDestination
+                    {
+                        Id = Guid.NewGuid(),
+                        Country = queryTransportOption.ToCountry,
+                        City = queryTransportOption.ToCity,
+                        Counter = 1
+                    };
+                    dbContext.PopularDestinations.Add(destination);
+                }
+                else
+                {
+                    destination.Counter++;
+                    dbContext.PopularDestinations.Update(destination);
+                }
+                
+                if (type == null)
+                {
+                    type = new PopularTransportType
+                    {
+                        Id = Guid.NewGuid(),
+                        Type = queryTransportOption.Type,
+                        Counter = 1
+                    };
+                    dbContext.PopularTransportTypes.Add(type);
+                }
+                else
+                {
+                    type.Counter++;
+                    dbContext.PopularTransportTypes.Update(type);
```

Comment:
I think it would work without calling Update since there's SaveChangesAsync after

Reply:
Fixed, removed it.